Tests/cleanup.py

```diff
@@ -26,15 +26,18 @@ def print_usage(file=sys.stderr):
     print("""Usage: cleanup.py [options]
 This tool cleans the cloud environment CLOUD by removing any resources whose name start with PREFIX.
 Options:
- [-c/--os-cloud OS_CLOUD] sets cloud environment (default from OS_CLOUD env)
- [-i/--prefix PREFIX] sets prefix (default from PREFIX env)
+ [-c/--os-cloud OS_CLOUD]  sets cloud environment (default from OS_CLOUD env)
+ [-p/--prefix PREFIX]      sets prefix to identify resources (default from PREFIX env)
+ [-i/--ipaddr addr[,addr]] list of IP addresses to identify ports to delete (def: delete all)
+                           the specified strings will be matched against the start of the addrs
 """, end='', file=file)
 
 
 class Janitor:
-    def __init__(self, conn, prefix=""):
+    def __init__(self, conn, prefix="", ipfilter=[]):
```

Comment:
Default values shouldn't be mutable. Better use a tuple here. Any list is also a tuple, so type-wise this is fine.

```suggestion
    def __init__(self, conn, prefix="", ipfilter=()):
```